contributor/es/04-mechanics-of-contributing-article-es.asciidoc

```diff
@@ -0,0 +1,158 @@
+
+==== Pruebas y automatización
+
+El punto básico aquí es permitir que el https://innersourcecommons.org/learn/learning-path/trusted-committer[_Trusted Committer_] valide la contribución sin tu presencia y asegurar una fácil mantenibilidad.
+Imagina que has construido una característica o el manejo de una rareza irresoluble, o un importante ajuste de rendimiento, y tu código no es del todo obvio (o incluso podría parecer hacky / incorrecto a primera vista).
+Si has cubierto esto con una prueba - e idealmente has arrojado algunas palabras sobre la razón de ser de la misma en un comentario - un futuro editor conseguirá recordar el propósito del código, y la(s) prueba(s) asegurará(n) que el valor que tu código realiza se mantendrá, incluso en las nuevas implementaciones.
+Para conseguirlo, haz lo siguiente:
+
+* Añade pruebas para tu contribución de código, para que la validación de la función de tu contribución por parte de otros funcione bien, incluso después de algún tiempo, cuando trabajes en otros proyectos o puedas haber dejado de contribuir a este proyecto.
+ ** A menudo, los proyectos tendrán comprobaciones automatizadas de las solicitudes de extracción utilizando esas pruebas y el nivel de cobertura del código. Intenta cumplir con los criterios que estas pruebas imponen.
+* Muchos proyectos proporcionarán scripts de construcción y validación del proyecto que le permiten probar localmente sus cambios.
+ ** Utilízalos para asegurarte de que tu contribución funciona lo mejor posible antes de abrir un pull request.
+ ** Tener que revisar solicitudes de extracción defectuosas con errores fáciles de arreglar a menudo molesta a los Trusted Committer. No arreglarán tu código pero te pedirán que lo hagas. Esto podría crear más viajes de ida y vuelta y ralentizar la fusión.
+ ** Sin embargo, nadie es perfecto. Haz lo mejor que puedas, utiliza scripts de validación preparados si los hay, y da lo mejor de ti con un pull request.
+ ** Si tu pull request sigue rompiendo pruebas, y no puedes averiguar por qué después de dar lo mejor de ti: intenta resaltar esas pruebas en el comentario del pull request, ilustra tu comprensión actual del problema y pide ayuda al respecto.
+* No te olvides de tu propio proyecto, que fue el que desencadenó tu contribución en primer lugar. Crea una compilación modificada del proyecto compartido con tus cambios y pruébala en tu propio proyecto que la consume.
+
+==== Documentación y revisabilidad
+
+Debes asegurarte de que tu pull request incluya cualquier actualización de la documentación que sea relevante para tus cambios.
+Si la documentación se encuentra en un lugar diferente, asegúrate de añadirla allí y enlazarla en tu pull request.
+
+Para que la revisión del código sea lo más fácil posible para el Trusted Committer u otras personas que lo revisen, intenta seguir estos consejos:
```

Comment:
Feels weird to not have the link as it always have it, also its missing in the original version
```suggestion
Para que la revisión del código sea lo más fácil posible para el https://innersourcecommons.org/learn/learning-path/trusted-committer[_Trusted Committer_] u otras personas que lo revisen, intenta seguir estos consejos:
```